libnavigation-base/src/test/java/com/mapbox/navigation/base/util/route/RouteUtilTest.kt

```diff
@@ -0,0 +1,164 @@
+                        provideBooleansProvider(false, true, false),
+                        false,
+                    ),
+                    arrayOf(
+                        "route closure at last waypoint, include closures = false",
+                        "route_closure_last_coordinate.json",
+                        provideBooleansProvider(true, true, false),
+                        provideBooleansProvider(true, true, false),
+                        true,
+                    ),
+                    arrayOf(
+                        "route closure at last waypoint, include closures = true",
+                        "route_closure_last_coordinate.json",
+                        provideBooleansProvider(false, false, true),
+                        provideBooleansProvider(false, false, true),
+                        false,
+                    ),
+                    arrayOf(
+                        "route closure between silent and regular waypoints",
+                        "route_closure_between_silent_and_regular_waypoints.json",
+                        provideBooleansProvider(true, true, true),
+                        provideBooleansProvider(true, true, true),
+                        true,
+                    ),
+                    arrayOf(
+                        "route closure between two regular waypoints",
+                        "route_closure_between_two_regular_waypoints.json",
+                        provideBooleansProvider(true, true, true),
+                        provideBooleansProvider(true, true, true),
+                        true,
+                    ),
+                )
+
+            private fun mockNavigationRoute(
+                routeRaw: String,
+                snappingIncludeClosures: BooleansProvider?,
+                snappingIncludeStaticClosures: BooleansProvider?,
+            ): NavigationRoute {
+                val route = DirectionsRoute.fromJson(
+                    FileUtils.loadJsonFixture(routeRaw),
+                    mockk {
+                        every {
+                            snappingIncludeClosuresList()
+                        } answers {
+                            snappingIncludeClosures?.invoke()
+                        }
+                        every {
+                            snappingIncludeStaticClosuresList()
+                        } answers {
+                            snappingIncludeStaticClosures?.invoke()
+                        }
+                        every { geometries() } returns DirectionsCriteria.GEOMETRY_POLYLINE6
+                    },
+                    "uuid"
+                )
+                return mockk {
+                    every { directionsRoute } returns route
+                    every { id } returns "-1"
+                }
+            }
+
+            private fun provideBooleansProvider(
+                vararg bool: Boolean?,
+            ): BooleansProvider = { bool.asList() }
+        }
```

Comment:
maybe that's simpler? 
```suggestion
            private fun mockNavigationRoute(
                directionResponse: String,
                snappingIncludeClosures: List<Boolean>?,
                snappingIncludeStaticClosures: List<Boolean>?,
            ): NavigationRoute {
                return createNavigationRoutes(
                    DirectionResponse.fromJson(directionResponse),
                    createRouteOptions(
                        snappingIncludeClosuresList = snappingIncludeClosures,
                        snappingIncludeStaticClosures = snappingIncludeStaticClosures,
                        TestSDKRouteParser(),
                        com.mapbox.navigation.base.route.RouterOrigin.Offboard
                ).first()
        }
```

P.S. I understand that given the time pressure you may don't have time to even discuss it, feel free to merge and get back to this question later 🙂

Reply:
yeah, it also would work 👍 let's keep the same, as we want to include this to release today